npm install fails with node Docker images

Hello,

Keystone 4.0.0-beta.8 has a regression where `npm install keystone` fails inside a `node:slim` Docker container. This issue was introduced by PR #4453, specifically the updated [knox dependency](https://github.com/keystonejs/keystone/blob/v4.0.0-beta.8/package.json#L54). It requires git for installation, which is not included in the production-focused node Docker images.

Workaround:

* add `apt-get install git` to the Dockerfile before `npm install`

Possible solutions:

* replace the `keystonejs/knox` dependency with [knox-s3](https://www.npmjs.com/package/knox-s3) (as discussed in Automattic/knox#326)
* publish `keystonejs/knox` to npm
* remove knox altogether, and replace with an alternative like [aws-sdk](https://www.npmjs.com/package/aws-sdk)

If using knox-s3 is good enough for now, I'd be happy to submit a PR. Longer term, eliminating knox seems wise.

@vitalbone, do you have an opinion?

Chad

---

### Expected behavior
`npm install keystone` should work when using the `node:slim` Docker image.

### Actual/Current behavior
It fails. This is the error with node 9.5.0 and npm 5.6.0.

    npm ERR! code ENOGIT
    npm ERR! No git binary found in $PATH
    npm ERR! 
    npm ERR! Failed using git.

For the sake of search indexing, earlier versions of npm also say:

    npm ERR! not found: git

### Steps to reproduce the actual/current behavior

1. Create a Dockerfile:

        FROM node:slim
        RUN npm install keystone

2. Build it:

        docker build .

### Environment

| Software         | Version
| ---------------- | -------
| Keystone         | 4.0.0-beta.8
| Node             | 9.5.0

--- a/admin/server/api/s3.js
+++ b/admin/server/api/s3.js
@@ -5,7 +5,7 @@ TODO: Needs Review and Spec
 module.exports = {
 
 	upload: function (req, res) {
-		var knox = require('knox');
+		var knox = require('knox-s3');
 		var keystone = req.keystone;
 		var Types = keystone.Field.Types;
 

--- a/fields/types/s3file/S3FileType.js
+++ b/fields/types/s3file/S3FileType.js
@@ -68,7 +68,7 @@ Object.defineProperty(s3file.prototype, 's3config', {
  */
 s3file.prototype.addToSchema = function (schema) {
 
-	var knox = require('knox');
+	var knox = require('knox-s3');
 	var field = this;
 
 	var paths = this.paths = {
@@ -333,7 +333,7 @@ s3file.prototype.generateHeaders = function (item, file, callback) {
  */
 s3file.prototype.uploadFile = function (item, file, update, callback) {
 
-	var knox = require('knox');
+	var knox = require('knox-s3');
 	var field = this;
 	var path = field.options.s3path ? field.options.s3path + '/' : '';
 	var prefix = field.options.datePrefix ? moment().format(field.options.datePrefix) + '-' : '';

--- a/package.json
+++ b/package.json
@@ -51,7 +51,7 @@
     "kerberos": "0.0.23",
     "keystone-storage-namefunctions": "1.1.1",
     "keystone-utils": "0.4.0",
-    "knox": "keystonejs/knox",
+    "knox-s3": "0.9.5",
     "less-middleware": "2.2.1",
     "letsencrypt-express": "2.0.6",
     "list-to-array": "1.1.0",

